app/Http/Controllers/BalanceItemsController.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\BalanceItem;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\DB;
+
+class BalanceItemsController extends Controller
+{
+    public function show()
+    {
+        $user_id = Auth::id();
+
+        //get all items for the user
+        $items = BalanceItem::select(DB::raw('id, label, entry_date, (amount / 100) as amount, DATE(entry_date) as date'))
+                              ->orderBy('entry_date', 'desc')
+                              ->where('user_id', '=', $user_id)
+                              ->get();
+
+        //get daily subtotals for the user
+        $groupedItems = BalanceItem::select(DB::raw('sum(amount / 100) as `amountSum`'), DB::raw('DATE(entry_date) as date'))
+                                     ->groupBy('date')
+                                     ->where('user_id', '=', $user_id)
+                                     ->orderBy('date', 'desc')
+                                     ->paginate(5);
+
+        //get total amount for the user
+        $sumOfItems = BalanceItem::select(DB::raw('sum(amount / 100) as `amountSum`'))
+                                   ->where('user_id', '=', $user_id)
+                                   ->get();
```

Comment:
Could you generate this data from the single query above?

Reply:
Yes, these two queries are just different aggregations of the data retrieved in the first query.

I could iterate through the data retrieved by the first query two create two objects (one for the daily subtotals, one for the grand total) in PHP before returning the data to the front end.

I could also do this in the front-end Vue component using JavaScript via a method/computed property after returning the data from the initial query.

In my experience I have got the best performance aggregating this data right in the MySQL queries and retuning them this way - especially as the data set gets larger but I am certainly open to a better way.